Added named routes

samir-plusb opened the pull request:
Hi,

I added named routes to the RoutingServiceProvider. It may be related to the following issue: #4 

Commit message:

Refactored the RoutingServiceProvider

- Implemented named Routes
- Added sanitization for named routes
- refactored the code
- changed the documentation and the composer.json

marcojanssen commented:
Very nice, thnx! Will look into it asap

marcojanssen commented on README.md:
```diff
@@ -1,5 +1,7 @@
 # RoutingServiceProvider #
 
+**Note**: This is just a fork of [marcojanssen/silex-routing-service-provider](https://github.com/marcojanssen/silex-routing-service-provider) with some adaptions for named routes. So all credits belong to [Marco Janssen](https://github.com/marcojanssen). And although Marco Janssen recommends to use it with php (not with yml/xml/etc), I recommend to use it with yaml, because I just tested it with ...
```
Could you revert this note? considering i'm adding it to my own repo

marcojanssen replied:
The reason way i recommend php is only because that can be opcached. Using yaml means you should cache it yourself which for a microframework as Silex shouldn't really be needed.

marcojanssen commented on README.md:
```diff
@@ -18,12 +20,12 @@
 - Install [Composer](http://getcomposer.org)
 
-- Add `marcojanssen/silex-routing-service-provider` to your `composer.json`:
+- Add `samir-plusb/silex-routing-service-provider` to your `composer.json`:
```
Same for this, i'll be more then happy to add you as contributor naturally

marcojanssen commented on README.md:
```diff
@@ -18,12 +20,12 @@
 {
     "require": {
-        "marcojanssen/silex-routing-service-provider": "1.2.*"
+        "samir-plusb/silex-routing-service-provider": "1.2.*"
```
And again, i'll up the version as well later on

marcojanssen commented on composer.json:
```diff
@@ -1,24 +1,34 @@
+      "name": "Marco Janssen",
+      "email": "dev@marcojanssen.net"
+    },{
+      "name": "Samir Rachidi",
```
:+1:

marcojanssen commented on composer.json:
```diff
@@ -1,24 +1,34 @@
-        "psr-0": {
-            "MJanssen\\": "src/"
-        }
+  "name": "samir-plusb/silex-routing-service-provider",
```
this should stay as is, otherwise packagist will mess up

marcojanssen commented:
Added some initial comments, will take a closer look at the code later on, want to make sure nothing breaks for backwards compatibility. Otherwise i have to bump up the major version number

marcojanssen commented on src/MJanssen/Provider/RoutingServiceProvider.php:
```diff
@@ -49,97 +50,121 @@ public function boot(Application $app)
-                $controller->requireHttps();
-            }
-        }
+        $this->sanitizeRouteName($route['routeName']);
```
Why do we need sanitation inside a validation method? I think you should call this from within addRoute, preferably without passing it by reference, that's in my personal opinion so easy to overlook

marcojanssen commented on src/MJanssen/Provider/RoutingServiceProvider.php:
```diff
@@ -49,97 +50,121 @@ public function boot(Application $app)
      * @throws \InvalidArgumentException
      */
-    protected function validateRoute($route)
+    protected function validateRoute(&$route)
```
This is passed by reference only because of the sanitation, which doesn't belong in this method anyway (validation is something entirely different then sanitation).

Could you move the sanitation method and also the pass by reference?

marcojanssen commented on src/MJanssen/Provider/RoutingServiceProvider.php:
```diff
@@ -49,97 +50,121 @@ public function boot(Application $app)
+     * @return string
      */
-    protected function getController(Application $app, $route)
+    protected function sanitizeRouteName(&$routeName)
```
If it's passed by reference why return it? Could you please just remove the pass by reference and return eouteName by default?

marcojanssen commented on src/MJanssen/Provider/RoutingServiceProvider.php:
```diff
@@ -49,97 +50,121 @@ public function boot(Application $app)
-        if(!in_array(strtolower($route['method']), $availableMethods)) {
-            throw new InvalidArgumentException('Method is not valid, only the following methods are allowed: get, put, post, delete');
-        }
+        if ($routeName) {
```
If it doesn't exist do a early return.

``` php
if (empty($routeName)) {
    return;
}

//sanitation

return $routeName;

```

marcojanssen commented on tests/MJanssen/Provider/RoutingServiceProviderTest.php:
```diff
@@ -30,12 +30,12 @@ public function testApplicationRoutes()
         $app = new Application();
```
I think your missing a test case asserting that the routename is actually what you specified

marcojanssen commented on src/MJanssen/Provider/RoutingServiceProvider.php:
```diff
@@ -49,97 +50,121 @@ public function boot(Application $app)
+     * @param array $route
      * @throws InvalidArgumentException
      */
     public function addRoute(Application $app, $route)
```
if $route should be a array why not enforce it as 
```
addRoute(Application $app, array $route)
```

marcojanssen commented:
Done with reviewing

samir-plusb commented:
Thank you for the valuable feedback! I made your requested changes and merged/rebased them into one commit (and updated the tag). I also added a new test case...

marcojanssen commented:
Looks good, will do some minor refactoring. Thanks for your contribution, will tag it this weekend
